### PR TITLE
added checks to prevent misc errors

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* None

*Description of changes:*
- added checks for valid nr.applicants object, which should fix this error:
   https://sentry.io/organizations/registries/issues/2098142767/events/90a90f68303243b29277a68f6233ec34/
- added check for valid summary.receipt object, which should fix this error:
   https://sentry.io/organizations/registries/issues/2098875580/events/332ddfeba28d47929f2eff9f516eb63c/
- added checks for valid querySelector function, which should fix these errors:
   https://sentry.io/organizations/registries/issues/2229040415/events/d3183f0ec1ad42fca073711a6bf19e2b/
   https://sentry.io/organizations/registries/issues/2227735742/events/77c9f37ef78f4a0d92db15c48f84986d/
   https://sentry.io/organizations/registries/issues/2227735759/events/a37fbf1afd984f08b5476f1fedde7351/

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

**Review by cameron-freshworks — APPROVED** [review]
LGTM!

File: client/src/components/payment/payment-summary.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <template>
   <v-expand-transition>
-    <!-- do not display until payments are fetched -->
-    <section class="payment-summary" v-if="summary">
+    <!-- check for valid receipt, as the payment may exist but not be completed -->
+    <section class="payment-summary" v-if="summary.receipt">
```
Comment (jordiwes) [inline client/src/components/payment/payment-summary.vue:4]: Quick question, just to be thorough, should this be v-if="summary && summary.receipt" ?

**Review by jordiwes — COMMENTED** [review]
Other than above comment LGTM

File: client/src/components/payment/payment-summary.vue (hunk 1 above)
Comment (severinbeauvais) [reply client/src/components/payment/payment-summary.vue:4]: Good question!

This component is in a v-for loop that iterates over summaries (and summary, a prop, is each item). So in theory this component is not rendered if the loop is empty, and summary is always defined. I tested a few scenarios locally to verify that this component doesn't have an undefined/null summary object.

Note that some components in Namerequest are pre-mounted and are then conditionally displayed. In that case, the component may exist without valid props. But not in this case (I think).

If you insist, I can add the original check back in. But I'd be more curious to see if it ever fails and then handle THAT situation :)

Comment (jordiwes) [reply client/src/components/payment/payment-summary.vue:4]: got it & do not insist ;)

**Review by jordiwes — APPROVED** [review]

**Pull request merged**